python/agents/bidi-demo/app/main.py

```
@@ -0,0 +1,213 @@
+    return FileResponse(Path(__file__).parent / "static" / "index.html")
+
+# ========================================
+# WebSocket Endpoint
+# ========================================
+
+@app.websocket("/ws/{user_id}/{session_id}")
+async def websocket_endpoint(websocket: WebSocket, user_id: str, session_id: str) -> None:
+    """WebSocket endpoint for bidirectional streaming with ADK."""
+    logger.debug(f"WebSocket connection request: user_id={user_id}, session_id={session_id}")
+    await websocket.accept()
+    logger.debug(f"WebSocket connection accepted")
```

Ruff (0.14.4)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)